symengine/expand.cpp

```diff
@@ -301,6 +301,51 @@ class ExpandVisitor : public BaseVisitor<ExpandVisitor>
             return;
         }
 
+        if (eq(*_base, *E) and !is_a_Number(*self.get_exp())) {
+            if (is_a<Add>(*self.get_exp())) {
+                RCP<const Basic> ret1 = one, ret2;
+                RCP<const Basic> coef
+                    = (down_cast<const Add &>(*self.get_exp())).get_coef();
+                ret2 = pow(E, coef);
+
+                for (auto &p :
+                     (down_cast<const Add &>(*self.get_exp())).get_dict()) {
+                    if (is_a<Log>(*p.first)) {
+                        RCP<const Number> s = rcp_static_cast<const Number>(
+                            down_cast<const Log &>(*p.first).get_arg());
+                        ret1 = mul(ret1, pownum(s, p.second));
+                    } else {
+                        ret2 = mul(ret2, pow(E, mul(p.first, p.second)));
```

Comment:
Repeatedly doing `mul` is inefficient. You should use the internal structures to speed this up.